File: app/src/main/java/com/abs/launcher/util/cache/Cache.java
package com.abs.launcher.util.cache;

import java.util.Map;

/**
 * Created by zy on 17-5-19.
 */

public interface Cache<K, T> {
    T get(K key);
    void put(K key, T value);
    T remove(K key);
    Map<K, T> getAll();
    void clear();
}

File: app/src/main/java/com/abs/launcher/util/cache/Cache.kt
package com.abs.launcher.util.cache

/**
 * Created by zy on 17-12-8.
 */
interface Cache<K, V> {
    fun put(key: K, value: V)
    fun get(key: K): V?
    fun remove(key: K): V?
    fun all(): Map<K, V> {
        return HashMap()
    }
}
